mkosi

```diff
@@ -564,6 +564,18 @@ def mount_image(args, workspace, loopdev, root_dev, home_dev, srv_dev, root_read
 
             umount(root)
 
+@complete_step("Assigning hostname")
+def assign_hostname(args, workspace):
+    root = os.path.join(workspace, "root")
+    hostname_path = os.path.join(root, "etc/hostname")
+    
+    if not args.hostname:
+        if os.path.isfile(hostname_path):
+            os.remove(hostname_path)
+    else:
+        with open(hostname_path, "w+") as f:
```

Comment:
uh, oh, i indeed was doing other stuff then... Sorry, I was very confused... Anyway the new patch looks great, and I merged it now.